### DISCUSSION TITLE
Unable to set breakpoints in VS code for debugging Theia extension

### DISCUSSION THREAD

**Discussion by umar-shabbir** [eclipse-theia/theia#13360] [Q&A]
I generated a Theia hello world sample extension using the [generator](https://github.com/eclipse-theia/generator-theia-extension)
Now I tried setting breakpoints in the file `hello-world-contribution.ts`
vscode `launch.json` is already generated and launching the debug session with `Start Browser Backend` works. However, the breakpoints are not binded and following message appears:
**Unbound breakpoint**
 **Some of your breakpoints could not be set. If you're having an issue, you can troubleshoot your launch configuration.**

From troubleshoot options, I tried searching the sources and there is no source map loaded for this file `hello-world-contribution.ts`
source maps are enabled in `tsconfig.json` and `launch.json`

I tried breakpoints on a VS code extension sample generated using the [generator](https://www.npmjs.com/package/generator-code). It works fine and breakpoints are activated. 

I also tried breakpoints on Theia blueprint and found same issue there. Unbound breakpoints.

What additional steps need to be done to enable debugging in Theia extensions?

**Comment by msujew** [comment]
Hey @umar-shabbir,

I assume the part of the extension you were trying to debug is in the `browser`section? In that case the code literally runs in your browser and not node - so vscode is unable to debug it (unless you use a very specific launch config - see below). Instead, we recommend to use your browsers debugging tools.

**Reply by sdirix** [reply]
For debugging the browser code within VS Code you can also use an additional launch configuration like [this one](https://github.com/eclipse-theia/theia/blob/9530193862783ca5197eb614f173a360061134d2/.vscode/launch.json#L121-L127).
